### PR TITLE
Remove DisablePreemption field from KubeSchedulerConfiguration

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind cleanup
/sig scheduling

**What this PR does / why we need it**:

With the refactoring of PostFilter extension point, DisablePreemption field in KubeSchedulerConfiguration is no longer needed as that can be deduced from profile's postFilterPlugins.

**Which issue(s) this PR fixes**:

/ref #91038 #89701.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/785-scheduler-component-config-api
```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -481,7 +467,7 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		// into the resources that were preempted, but this is harmless.
 		nominatedNode := ""
 		if fitError, ok := err.(*core.FitError); ok {
-			if sched.DisablePreemption || !prof.HasPostFilterPlugins() {
+			if !prof.HasPostFilterPlugins() {
 				klog.V(3).Infof("Pod priority feature is not enabled or preemption is disabled by scheduler configuration." +
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go:471]: Update the log line here. It doesn't seem like Pod priority affect this anymore.

File: test/integration/scheduler/util.go [hunk 2]
```diff
@@ -88,9 +89,19 @@ func initTest(t *testing.T, nsPrefix string, opts ...scheduler.Option) *testutil
 // initTestDisablePreemption initializes a test environment and creates master and scheduler with default
 // configuration but with pod preemption disabled.
 func initTestDisablePreemption(t *testing.T, nsPrefix string) *testutils.TestContext {
+	prof := schedulerconfig.KubeSchedulerProfile{
+		SchedulerName: v1.DefaultSchedulerName,
+		Plugins: &schedulerconfig.Plugins{
+			PostFilter: &schedulerconfig.PluginSet{
+				Disabled: []schedulerconfig.Plugin{
+					{Name: "*"},
```
Comment (alculquicondor) [inline test/integration/scheduler/util.go:97]: Let's use the name of the plugin, in case we have another default PostFilter plugin in the future.

**Comment by alculquicondor** [comment]
other than the nits, looks good.

/assign @liggitt 
for api review

File: test/integration/scheduler/util.go (hunk 2 above)
Comment (Huang-Wei) [reply test/integration/scheduler/util.go:97]: Done.

File: pkg/scheduler/scheduler.go (hunk 1 above)
Comment (Huang-Wei) [reply pkg/scheduler/scheduler.go]: Good catch. Updated.

File: pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -481,9 +467,8 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		// into the resources that were preempted, but this is harmless.
 		nominatedNode := ""
 		if fitError, ok := err.(*core.FitError); ok {
-			if sched.DisablePreemption || !prof.HasPostFilterPlugins() {
-				klog.V(3).Infof("Pod priority feature is not enabled or preemption is disabled by scheduler configuration." +
-					" No preemption is performed.")
+			if !prof.HasPostFilterPlugins() {
+				klog.V(3).Infof("No PostFilter plugins are registered, so no preemption will be performed.")
```
Comment (liggitt) [inline pkg/scheduler/scheduler.go:471]: This is preexisting, and maybe I'm misunderstanding, but this makes it seem like you have to enable postfilter plugins for preemption to work. That's pretty surprising to me.

**Comment by liggitt** [comment]
API bits look good. Since v1beta1 scheduler config is new this release, the release note doesn't need to say something was removed from v1beta1, this should be added to the docs detailing the differences between v1alpha2 and v1beta1

**Comment by liggitt** [comment]
/approve

for scheduler API change
/retest

File: pkg/scheduler/scheduler.go (hunk 3 above)
Comment (Huang-Wei) [reply pkg/scheduler/scheduler.go:471]: > this makes it seem like you have to enable postfilter plugins for preemption to work.

True, it's enabled by default. The background is that scheduler have almost "pluginize" all hard-coded logic into plugins. In 1.19 the defaultpreemption plugin follows that direction.

So for a default scheduler configuration, preemption functions the same as before. But users can still disable preemption by tweaking the config, just like what they did by specifying `DisablePreemption: false`.

**Comment by Huang-Wei** [comment]
Thanks @alculquicondor @liggitt for the review!

> Since v1beta1 scheduler config is new this release, the release note doesn't need to say something was removed from v1beta1, this should be added to the docs detailing the differences between v1alpha2 and v1beta1

Release note removed.

@alculquicondor would you mind adding a bullet to your PR https://github.com/kubernetes/website/pull/21437, in docs/reference/scheduling/config.md#beta-changes?

```
- `.disablePreemption` was removed. Users can disable preemption by disabling the "defaultpreemption" PostFilter plugin.
```

**Pull request merged**